src/Toml/Codec/Combinator/Custom.hs

```diff
@@ -46,3 +52,89 @@ nullary sum types.
 enumBounded :: (Bounded a, Enum a, Show a) => Key -> TomlCodec a
 enumBounded = match _EnumBounded
 {-# INLINE enumBounded #-}
+
+{- | Codec that checks the 'BiMap' on the given predicate.
+The predicate function returns the value, if the validation is successful, or
+the 'Text' of the error that should be returned in case of validation failure.
+
+__Example:__
+
+Let's imagine that we want to have the 'Text' field in @TOML@ that could only
+have 3 chars in it. In this case, you can write the following codec:
+
+@
+allEven :: ['Int'] -> 'Either' 'Text' ['Int']
+allEven xs =
+    if all even xs
+    then 'Right' xs
+    else 'Left' "This is wrong, I asked you for even only :("
+
+allEvenCodec :: TomlCodec [Int]
+allEvenCodec = Toml.'validate' allEven (Toml._Array Toml._Int) \"myEvenList\"
+@
+
+Then in your @TOML@ file you can have:
+
+@
+myEvenList = [2, 4, 6]
+@
+
+But the following one will lead to the error:
+
+@
+myEvenList = [1, 2, 3]
+@
+
+@
+tomland decode error:  This is wrong, I asked you for even only :(
+@
+
+@since 1.3.0.0
+-}
+validate :: (a -> Either Text a) -> TomlBiMap a AnyValue -> Key -> TomlCodec a
+validate p bimap = match (_Validate p bimap)
+{-# INLINE validate #-}
+
+{- | Similar to 'validate' but takes the predicate that returnes 'Bool'.
+The error in case of the validation failure looks like this:
+
+@
+Value does not pass the validation for key: KEY_NAME
+@
+
+__Example:__
+
+Let's imagine that we want to have the 'Text' field in @TOML@ that could only
+have 3 chars in it. In this case, you can write the following codec:
+
+@
+my3charTextCodec :: TomlCodec Text
+my3CharTextCodec = Toml.'validateIf' ((==) 3 . Text.length) Toml._Text "myKeyName"
+@
+
+The in your @TOML@ file you can have:
+
+@
+myKeyName = "abc"
+@
+
+But the following one will lead to the error:
+
+@
+myKeyName = "I have more than enough"
+@
+
+@
+tomland decode error:  Value does not pass the validation for key: myKeyName
+@
+
+@since 1.3.0.0
+-}
+validateIf :: forall a . (a -> Bool) -> TomlBiMap a AnyValue -> Key -> TomlCodec a
+validateIf p bimap k = validate validateEither bimap k
```

Comment:
Let's name the variable `biMap` not to confuse with `bimap` from `Data.Bifunctor`
```suggestion
validateIf p biMap k = validate validateEither biMap k
```